docs/src/architecture/assets.md

```diff
@@ -0,0 +1,61 @@
+# Assets
+The Miden Rollup aims to provide a reliable protocol and software for users to exchange assets. In Miden, users can create and trade arbitrary fungible and non-fungible assets. 
+
+## Intro
+We differentiate between native and non-native assets in Miden. Native assets follow the Miden asset model. Non-native assets are all other data structures of value that can be exchanged.   
+
+Recording of native assets in Polygon Miden suffices four goals:
+
+* Asset exchange should be parallelizable 
+* Asset ownership should be private 
+* Asset usage should be indeed censorship resistant
+* Fees can be paid using any asset
+
+All native assets in Miden are stored directly in accounts, like Ether in Ethereum. Miden does not track ownership of assets using global hashmaps, e.g., ERC20 contracts. Storage of assets locally in accounts provides privacy and the ability for client-side proofs. That is because ownership changes always only involve one account and not the change of a global hashmap. Thus, they can happen in parallel. Additionally, asset exchange is censorship resistant at this level because there is no global contract the transfer must pass through. Finally, users can pay fees in any asset.
+
+## Native assets
+Native assets are data structures issued by asset accounts and can be stored in account and note vaults. All native assets are encoded using a single `Word` (4 field elements), so it is easy to determine the type of an asset inside and outside Miden VM. The asset encodes both the ID of the issuing account and asset details. So, when we look at an asset, we know who issued it. It is guaranteed that there can never be a collision between a fungible and a non-fungible asset.
+
+### Issuance
+Specialized asset accounts called faucets issue assets by creating notes. Like any other account, faucets have an Account ID called `faucet_id`. Users can define specific properties of the asset when creating the faucet. Asset creation is similar to creating ERC20 contracts on Ethereum. The rules for who can issue assets and how are fully customizable and defined by the code deployed in the faucet account. The only restrictions are:
```

Comment:
Could you clarify what you mean here by "can define specific properties of the asset"?

Reply:
I agree with @frisitano - it would be good to expand and clarify here. Maybe try something like this:

Only specialized accounts called *faucets* can issue assets. Just like with regular accounts, anyone can create a faucet account. The code in these accounts defines rules for how assets can be minted, who can mint them etc. Conceptually, faucet accounts on Miden are similar to ERC20 contracts on Ethereum, but Miden guarantees certain asset properties at the protocol level. Specifically ...

Also, we should probably add a link here to the accounts section where we describe different types of accounts.

Reply:
Oh - and we should also mention that faucets can issue only one type of assets (e.g., either fungible or non-fungible - but it can't issue both).

Also, not sure if it is worth mentioning, but could be interesting: just like regular accounts, faucet accounts can store assets issued by other faucet accounts.

Reply:
nice, I used your suggestion